serving/ingress/pkg/reconciler/ingress/resources/route.go

```diff
@@ -144,9 +144,12 @@ func makeRoute(ci *networkingv1alpha1.Ingress, host string, rule networkingv1alp
 
 	// Target the HTTPS port and configure passthrough when:
 	// * the passthrough annotation is set.
-	// * the ingress.spec.tls is set. (DomainMapping with BYP cert.)
-	// * the internal-encryption is enabled.
-	if _, ok := annotations[EnablePassthroughRouteAnnotation]; ok || len(ci.Spec.TLS) > 0 || isInternalEncryptionEnabled(rule) {
+	// * the ingress.spec.tls is set for an external domain (e.g. DomainMapping with BYP cert.)
+	// * the destination service uses a https port.
+	if _, ok := annotations[EnablePassthroughRouteAnnotation]; ok ||
+		len(ci.GetIngressTLSForVisibility(networkingv1alpha1.IngressVisibilityExternalIP)) > 0 ||
```

Comment:
Was not `len(ci.Spec.TLS) > 0` enough in the past?

Reply:
Yes, but now we also have TLS for cluster-local-domains (visibility = cluster-local), then we should not do passthrough as the "external-gateway" part then still uses http. This basically filters out the new thing to be the same as before.

Reply:
@ReToCode what is needed to access the service if someone wants to use ca verification at the client side?
Does he need to get the certificate (self-signed in this case) from the corresponding ns? What happens if someone replaces the ingress cert as in here: https://docs.openshift.com/container-platform/4.15/security/certificates/replacing-default-ingress-certificate.html?

Reply:
Depends on from where:
- Caller is outside of the cluster, it needs to trust the OCP ingress CA (the one you linked). Out of scope of Knative.
- Caller is inside the cluster, it needs to trust the CA that they put in cert-manager. As described in the docs.

Reply:
I think the confusion comes from this:
- The pass-through here is only relevant for `system-internal-tls`, there we have some code in S-O that copies the OCP ingress cert (the same as OCP routers use) to kourier to make it still work. Not really sure why we did that in the first place, but it works for now.

Reply:
> Caller is outside of the cluster, it needs to trust the OCP ingress CA (the one you linked). Out of scope of Knative.

If it is passthrough from what I read the same OCP cert should be presented at the Kourier ingress side for a TLS connection. no?

Reply:
> If it is passthrough from what I read the same OCP cert should be presented at the Kourier ingress side for a TLS connection, no? Thus the copy. @mgencur any background info on this one?

What backgroung info do you mean?

The certificate for OpenShift ingress is copied to the net-kourier-controller [here](https://github.com/openshift-knative/serverless-operator/pull/2663/files#diff-f2a6e099ca204caa0f720b4b7ac02b8f2e3f0d47029f9aa26ca95e24cc74fbadR104) so the end user can trust the OCP ingress certificate without change and still use the internal-encryption.

I think this is to simplify the process for the user as the user doesn't have to setup trust to any new certificates presented by the Kourier gateway. And Kourier gateway doesn't have to generate new certificates.

I guess this could be also achieved by generating new certificates for Kourier, then using a re-encrypt Route for Knative services and the certificate from Kourier could be automically setup for the Route.

Reply:
Using a re-encrypt route would involve one aditional decryption+encryption on the data path.

Reply:
+ we discussed yesterday, not sure if this helps with the http2 case. With passthrough, haproxy does not touch the http connection. I assume with reencrypt it would be touched, which could break it (I haven't tested that though).